conda/common/configuration.py

```diff
@@ -1347,32 +1316,86 @@ def __init__(cls, name, bases, attr):
         )
 
 
+CONDARC_FILENAMES = (".condarc", "condarc")
+YAML_EXTENSIONS = (".yml", ".yaml")
+
+
 class Configuration(metaclass=ConfigurationType):
-    def __init__(self, search_path=(), app_name=None, argparse_args=None):
+    def __init__(self, search_path=(), app_name=None, argparse_args=None, **kwargs):
         # Currently, __init__ does a **full** disk reload of all files.
         # A future improvement would be to cache files that are already loaded.
         self.raw_data = {}
         self._cache_ = {}
         self._reset_callbacks = IndexedSet()
         self._validation_errors = defaultdict(list)
 
-        self._set_search_path(search_path)
+        self._set_search_path(search_path, **kwargs)
         self._set_env_vars(app_name)
         self._set_argparse_args(argparse_args)
 
-    def _set_search_path(self, search_path):
-        self._search_path = IndexedSet(search_path)
-        self._set_raw_data(load_file_configs(search_path))
+    @staticmethod
+    def _expand_search_path(
+        search_path: Iterable[Path | Template | str],
+        **kwargs,
+    ) -> Iterable[Path]:
+        for search in search_path:
+            # use string.Template instead of os.path.expand so additional variables can be passed
```

Comment:
There is an issue with using [`string.Template`](https://docs.python.org/3/library/string.html#template-strings) here instead since `os.path.expandvars` [has this in addition](https://docs.python.org/3/library/os.path.html#os.path.expandvars):

> On Windows, `%name%` expansions are supported in addition to `$name` and `${name}`.

In other words, we'll have to somehow retrofit this to `string.Template` if we want to switch to it.

Reply:
Maybe this would suffice?

```python
class WindowsTemplate(string.Template):
    delimiter = "%"
```

From the docs:

> Advanced usage: you can derive subclasses of [Template](https://docs.python.org/3/library/string.html#string.Template) to customize the placeholder syntax, delimiter character, or the entire regular expression used to parse template strings. To do this, you can override these class attributes:
> 
> delimiter – This is the literal string describing a placeholder introducing delimiter. The default value is $. Note that this should not be a regular expression, as the implementation will call [re.escape()](https://docs.python.org/3/library/re.html#re.escape) on this string as needed. Note further that you cannot change the delimiter after class creation (i.e. a different delimiter must be set in the subclass’s class namespace).

Reply:
Unfortunately `string.Template` doesn't support `%...%` without extra work. Further more `string.Template` treated escaped delimiters differently than `os.path.expandvars` does:

```
# on all OSes
assert string.Template("$$").safe_substitute() == "$"

# on Unix
assert os.path.expandvars("$$") == "$$"

# on Windows
assert os.path.expandvars("$$") == "$"
```

Because of this I ended up implementing a new function that properly handles `$VAR`, `${VAR}`, and `%VAR%` and leaves all other patterns as-is.